src/aliased_buffer.h

```diff
@@ -26,27 +29,69 @@ namespace node {
  * The encapsulation herein provides a placeholder where such writes can be
  * observed. Any notification APIs will be left as a future exercise.
  */
+
+enum class AliasedBufferType {
+  kNew,
+  kCopy,
+  kShared,
+  kAssigned,
+  kToBeDeserialized,
+  kDeserialized
+};
+
 template <class NativeT,
           class V8T,
           // SFINAE NativeT to be scalar
-          typename = std::enable_if_t<std::is_scalar<NativeT>::value>>
+          typename Trait = std::enable_if_t<std::is_scalar<NativeT>::value>>
 class AliasedBufferBase {
  public:
-  AliasedBufferBase(v8::Isolate* isolate, const size_t count)
-      : isolate_(isolate), count_(count), byte_offset_(0) {
-    CHECK_GT(count, 0);
+  static std::unordered_set<AliasedBufferBase*> buffers;
```

Comment:
Fwiw, this needs to be per-Environment, otherwise it’s subject to race conditions.

(But as you know, I don’t think this is a good idea anyway.)

Reply:
I guess we could guard this behind a flag (say when it's used in mksnapshot). This is only done for debugging anyway (in case there are unknown AliasedArrays not serialized)

Reply:
I removed this for the time being, it needs more flags to avoid races, but it's just for debugging, and not essential for snapshotting.